exchange/filecoinretrieval/filecoinretrieval.go

```diff
@@ -85,12 +99,28 @@ func (fe *FilecoinExchange) subscriber(event datatransfer.Event, channelState da
 		return
 	}
 	fe.transfersLk.RUnlock()
+	if tf.events == nil {
+		return
+	}
+
+	// TODO: are these the correct mappings?
```

Comment:
I think what we want to listen to is:
* `datatransfer.Completed` for `exchange.SuccessEvent`
* `datatransfer.Failed, datatransfer.Cancelled` for `exchange.FailureEvent`

Not sure about the rest.

Reply:
OK Looks like the subscriber will receive two sets of events. One about the [channel events](https://github.com/filecoin-project/go-data-transfer/blob/master/events.go) and one about [status of a channel](https://github.com/filecoin-project/go-data-transfer/blob/master/statuses.go). 

We probably want to listen to one set of those?

Cc @hannahhoward

Reply:
@willscott  see Hannah's comment here:
- https://github.com/filecoin-project/go-legs/pull/72#pullrequestreview-865604508

Looks like some thought is needed to emit `ErrorEvent` correctly here, which is meant to signal _non-terminal_ temporary failures?

Reply:
i think we do need to handle error events, or we end up in cases like:
```
2022-01-31T09:29:37.183+0100	DEBUG	dt-impl	impl/events.go:272	channel QmfCi3WzLCYvuUd8f5LS6YMPG3QaxKacpYAbobqkSFv537-12D3KooWE8yt84RVwW3sFcd6WMjbUdWrZer2YtT4dmtj3dHdahSZ-1643617771875890002 had transport send error: cant open message sender to peer 12D3KooWE8yt84RVwW3sFcd6WMjbUdWrZer2YtT4dmtj3dHdahSZ: failed to dial 12D3KooWE8yt84RVwW3sFcd6WMjbUdWrZer2YtT4dmtj3dHdahSZ:
  * [/ip4/192.168.0.109/tcp/39383] dial backoff
2022-01-31T09:29:37.183+0100	DEBUG	data-transfer	channels/channels.go:395	send data transfer event	{"name": "SendDataError", "transfer ID": 1643617771875890002, "args": [{}]}
2022-01-31T09:29:37.183+0100	DEBUG	gs-notifications	notifications/publisher.go:233	processing notification command	{"cmd": {}, "remaining in queue": 0}
2022-01-31T09:29:37.183+0100	DEBUG	data-transfer	channels/channels.go:116	process data transfer listeners	{"name": "SendDataError", "transfer ID": 1643617771875890001}
2022-01-31T09:29:37.183+0100	DEBUG	filecoin_retrieval	filecoinretrieval/filecoinretrieval.go:185	unrecognized data transfer event: SendDataError
2022-01-31T09:29:37.183+0100	DEBUG	data-transfer	channels/channels.go:116	process data transfer listeners	{"name": "SendDataError", "transfer ID": 1643617771875890002}
2022-01-31T09:29:37.183+0100	DEBUG	filecoin_retrieval	filecoinretrieval/filecoinretrieval.go:185	unrecognized data transfer event: SendDataError
```

where we're totally unable to connect but we're not relaying that back to the board that we should be trying to start up other options